SteamProject/MySteam/Steam.cpp

```diff
@@ -155,87 +143,133 @@ void FSteam::OpenCategoriesMenu()
 		system("cls");
 		std::cout << "-----------------Cateories Menu-----------------" << std::endl;
 		std::cout << "\t\t1 - Add category \n\t\t2 - Delete category \n\t\or Press Enter to return to the Main menu \n" << std::endl;
+
 		std::cin.clear();
 		std::cin.ignore(1000, '\n');
 		if (std::cin.peek() == '\n')
 		{
 			break;
 		}
-		CategoriesOption = ValidateInput("choose an option");
+		CategoriesOption = ValidateInput("");
 
-		if (CategoriesOption == 1)
+		switch (CategoriesOption)
 		{
-			if (MainHandler.HasSpace())
-			{
-				system("cls");
-				std::cout << "Enter the name of the category >> ";
-				std::string NewCategoryName;
-				std::cin >> NewCategoryName;
-
-				if (MainHandler.AddCategory(NewCategoryName))
-				{
-					std::cout << "Category added!" << std::endl;
-					std::cin.ignore();
-					std::cin.get();
-				}
-				else
-				{
-					std::cout << "Invalid name!" << std::endl;
-					std::cin.ignore();
-					std::cin.get();
-					CategoriesOption = 0;
-				}
-			}
-			else
-			{
-				std::cout << "You don't have more space for new categories!" << std::endl;
-				std::cin.ignore();
-				std::cin.get();
-				CategoriesOption = 0;
-			}
-		} 
-		else if (CategoriesOption == 2)
-		{
-			if (MainHandler.GetCategoriesAmount() == 0)
-			{
-				std::cout << "There are no categories saved." << std::endl;
-				std::cin.ignore();
-				std::cin.get();
-			}
-			else
-			{
-				do
-				{
-					system("cls");
-					MainHandler.ShowCategories();
-					int SelectedCategoryIndex = ValidateInput("Choose a category");
-					if (SelectedCategoryIndex >= 0 && SelectedCategoryIndex < MainHandler.GetCategoriesAmount())
-					{
-						MainHandler.DeleteCategory(SelectedCategoryIndex);
-						break;
-					}
-				} while (true);
-			}
-		}
-		else
-		{
-			CategoriesOption = 0;
-		}
+		case 1:
+			AddCategoryMenu();
+			break;
 
+		case 2:
+			DeleteCategoryMenu();
+			break;
+		}
```

Comment:
Very well done here! Look at how much cleaner and easier to read this all looks by breaking it up into smaller functions!